sdk/core/azure-core/src/samples/java/com/azure/core/http/rest/PagedIterableBaseJavaDocCodeSnippets.java

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.rest;
+
+import java.util.Iterator;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import reactor.core.publisher.Mono;
+
+/**
+ * Code snippets for {@link PagedIterableBase}
+ */
+public class PagedIterableBaseJavaDocCodeSnippets {
+
+    static class CustomPagedFlux<String> extends PagedFluxBase<String, PagedResponse<String>> {
+        CustomPagedFlux(Supplier<Mono<PagedResponse<String>>> firstPageRetriever) {
+            super(firstPageRetriever);
+        }
+
+        CustomPagedFlux(Supplier<Mono<PagedResponse<String>>> firstPageRetriever,
+            Function<java.lang.String, Mono<PagedResponse<String>>> nextPageRetriever) {
+            super(firstPageRetriever, nextPageRetriever);
+        }
+    }
+
+    /**
+     * Provides an example for iterate over each response using streamByPage function.
+     **/
+    public void streamByPageSnippet() {
+        // BEGIN: com.azure.core.http.rest.pagedIterableBase.streamByPage
+        // process the streamByPage
+        CustomPagedFlux<String> customPagedFlux = createCustomInstance();
+        PagedIterableBase<String, PagedResponse<String>> customPagedIterableResponse =
+            new PagedIterableBase<>(customPagedFlux);
+        customPagedIterableResponse.streamByPage().forEach(resp -> {
+            System.out.printf("Response headers are %s. Url %s  and status code %d %n", resp.getHeaders(),
```

Comment:
nit: anyway to avoid the same exact code being written 3x in this file?  otherwise lgtm

Reply:
These are code snippets that get inserted into our Javadocs. They all have unique codesnippet tags and get inserted for different methods. So, in this case, when viewed on javadocs will provide a clear self-contained sample at the cost of a bit of duplication.